client/components/async-load/index.jsx

```diff
@@ -31,8 +39,11 @@ export default class AsyncLoad extends Component {
 	}
 
 	require() {
-		this.props.require( ( component ) => {
-			this.setState( { component } );
+		const requireFunction = this.props.require;
+		requireFunction( ( component ) => {
+			if ( this.state.require === requireFunction ) {
```

Comment:
Do we need `require` to be in state? Isn't it enough to compare the current prop value to the `requireFunction` saved in the closure above? i.e. `this.props.require === requireFunction`